vendor/bat-native-ads/src/bat/ads/internal/notifications.cc

```diff
@@ -49,33 +49,42 @@ void Notifications::Initialize(InitializeCallback callback) {
 bool Notifications::Get(const std::string& id, NotificationInfo* info) const {
   DCHECK(is_initialized_);
 
-  auto notification = notifications_.find(id);
-  if (notification == notifications_.end()) {
+  auto iter = std::find_if(notifications_.begin(), notifications_.end(),
+      [&id](const auto & v){return v.id == id;});
+  if (iter == notifications_.end()) {
     return false;
   }
 
-  *info = notification->second;
+  *info = *iter;
 
   return true;
 }
 
 void Notifications::Add(const NotificationInfo& info) {
   DCHECK(is_initialized_);
 
-  notifications_.insert({info.id, info});
+  notifications_.push_back(info);
 
   SaveState();
 }
 
+void Notifications::Remove() {
+  if (!notifications_.empty()) {
```

Comment:
Can we change this to if empty then return and move the pop to end of function

Reply:
don't agree

Reply:
It is good practice to fail fast